lib/fauna/errors.rb

```diff
@@ -43,7 +63,7 @@ def self.raise_for_status_code(request_result)
     #            :: A simple string as the message.
     def initialize(request_result)
       @request_result = request_result
-      errors_raw = request_result.response_content[:errors]
+      errors_raw = InvalidResponse.get_or_invalid request_result.response_content, :errors
```

Comment:
I don't like raising an error while building an error exception, as it can mask what's going on. I think in the case this isn't valid, we could just pass errors directly through (or not at all). Not all http responses are going to come from FaunaDB in the real world.

If we do that I think an implementation of it could be just to wrap the constructing of the errors field here and if something happens we handle it here. Then we can properly signal up from `ErrorData` and friends if they failed to parse something without needing some internal parse error exception thing.